### ISSUE TITLE
Colored Coin and UnColored Coin scale is different in listunspent

### ISSUE THREAD

**Issue by nakajo2011** [chaintope/tapyrus-core#146]
# Describe
Colored Coin and UnColored Coin(TPC) amounts scale is different. Colored Coin scale is 'tapyrus' unit, but UnColored Coin scale is 'TPC' unit in `listunspent`.

# Expected Work
I expected to Colored Coin scale is also 'TPC' unit that is same as UnColored Coin  in `listunspent`.

# Actual Work
`listunspent` result is bellow:
```
$ docker-compose exec tapyrusd tapyrus-cli -conf=/etc/tapyrus/tapyrus.conf -rpcwallet="wallet-954cfd3f3a72ee44426c57c9419723ab" listunspent
[
  {
    "txid": "d252536de2def8f1afdb8232dae75a629b372f186ef89740a40f7d20d34c8245",
    "vout": 0,
    "address": "22VNVDjEMrXNZv1RyM3HpzbDXV7fk34Y3M2dZ3q4KrWXxjHkiDY97nsTAEBaUBixZZogR5fhfEeDQk8N",
    "token": "c1c592fb1a83738f48528446bcdfc2cb0538fa42ced2c66aba47f8de8091700077",
    "amount": 100,
    "scriptPubKey": "21c1c592fb1a83738f48528446bcdfc2cb0538fa42ced2c66aba47f8de8091700077bc76a91444b417de725005265a80e4aa17b65df9a94707ed88ac",
    "confirmations": 2,
    "spendable": true,
    "solvable": true,
    "safe": true
  },
  {
    "txid": "4059dbaff1e666b0e9c3d4218b31d49bcdc552d00ca2e154d849a58131501457",
    "vout": 1,
    "address": "n1zCGdCQtvp8GxQWPtm82CBQbZpbwqqDKd",
    "token": "TPC",
    "amount": 9.99860000,
    "scriptPubKey": "76a914e08969a5d894692f72b23ac60d8b85eb7709b84188ac",
    "confirmations": 1,
    "spendable": true,
    "solvable": true,
    "safe": true
  },
```

but `getrawtransaction` result of `d252536de2def8f1afdb8232dae75a629b372f186ef89740a40f7d20d34c8245` is bellow:
```
$ docker-compose exec tapyrusd tapyrus-cli -conf=/etc/tapyrus/tapyrus.conf getrawtransaction d252536de2def8f1afdb8232dae75a629b372f186ef89740a40f7d20d34c8245 1
{
  "txid": "d252536de2def8f1afdb8232dae75a629b372f186ef89740a40f7d20d34c8245",
  "hash": "142d27ec6ab810f4ad55f2b96bb3cf4df02086925099705ee051d9168e7a4400",
  "features": 1,
  "size": 226,
  "vsize": 226,
  "weight": 904,
  "locktime": 0,
  "vin": [
    {
      "txid": "dc15b95a480b1a6a5e9a4695a998d57c78f83337df4ce92e93b2342367245763",
      "vout": 0,
      "scriptSig": {
        "asm": "3044022028cd7e3d5344102591c2b60fd0c76fdd790c9b7d0f1505d0317023c6c74b881b0220537606807634168415b2ea521e998ade4ce18579dbc6aa19cdf4e5dd487f485c[ALL] 022fff854061c0b308906fd8b416238b4ad6d1b7f8abd7f75fab7909f2c85e9f77",
        "hex": "473044022028cd7e3d5344102591c2b60fd0c76fdd790c9b7d0f1505d0317023c6c74b881b0220537606807634168415b2ea521e998ade4ce18579dbc6aa19cdf4e5dd487f485c0121022fff854061c0b308906fd8b416238b4ad6d1b7f8abd7f75fab7909f2c85e9f77"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00000100,
      "n": 0,
      "scriptPubKey": {
        "asm": "c1c592fb1a83738f48528446bcdfc2cb0538fa42ced2c66aba47f8de8091700077 OP_COLOR OP_DUP OP_HASH160 44b417de725005265a80e4aa17b65df9a94707ed OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "21c1c592fb1a83738f48528446bcdfc2cb0538fa42ced2c66aba47f8de8091700077bc76a91444b417de725005265a80e4aa17b65df9a94707ed88ac",
        "reqSigs": 1,
        "type": "coloredpubkeyhash",
        "addresses": [
          "22VNVDjEMrXNZv1RyM3HpzbDXV7fk34Y3M2dZ3q4KrWXxjHkiDY97nsTAEBaUBixZZogR5fhfEeDQk8N"
        ]
      }
    }
  ],
  "hex": "0100000001635724672334b2932ee94cdf3733f8787cd598a995469a5e6a1a0b485ab915dc000000006a473044022028cd7e3d5344102591c2b60fd0c76fdd790c9b7d0f1505d0317023c6c74b881b0220537606807634168415b2ea521e998ade4ce18579dbc6aa19cdf4e5dd487f485c0121022fff854061c0b308906fd8b416238b4ad6d1b7f8abd7f75fab7909f2c85e9f77ffffffff0164000000000000003c21c1c592fb1a83738f48528446bcdfc2cb0538fa42ced2c66aba47f8de8091700077bc76a91444b417de725005265a80e4aa17b65df9a94707ed88ac00000000",
  "blockhash": "d43f959b83f3561e8657f29dbc7b12c326327e661b1927abddee59f1d9814cf5",
  "confirmations": 2,
  "time": 1626014860,
  "blocktime": 1626014860
}
```

# How to reproduce this issue
I used `Glueby` for easy issue token.

```
$ docker-compose up -d

$ docker-compose exec web bundle exec rails c
irb(main):001:0> alice = Glueby::Wallet.create
=> #<Glueby::Wallet:0x000055d24d734590 @internal_wallet=#<Glueby::Internal::Wallet:0x000055d24d734770 @id="6ee3e1144fdd4a34c0507b14030c4fc8">>
@id="22747903aa65248fa75251db202dc018">>
irb(main):023:1* Glueby::Internal::RPC.perform_as("wallet-"+alice.id) do |client|
irb(main):024:1*       client.generate(1, authority_key)
irb(main):025:0> end
=> ["6d8f1b23b339bc3a79bdf46d4840cbd12bf0df295f34b2dc6a32433c9fd29b36"]
irb(main):028:0> tokens = Glueby::Contract::Token.issue!(issuer: alice, amount: 100)
   (13.2ms)  SELECT sqlite_version(*)
  TRANSACTION (0.1ms)  begin transaction
  TRANSACTION (0.1ms)  SAVEPOINT active_record_1
  Glueby::Contract::AR::ReissuableToken Create (48.1ms)  INSERT INTO "glueby_reissuable_tokens" ("color_id", "script_pubkey", "created_at", "updated_at") VALUES (?, ?, ?, ?)  [["color_id", "c1dc28d2651ee01745b1f112c0502db47434ebc5e2140b7ba2b324ca3ada3655b3"], ["script_pubkey", "76a9148147ad309a550f723439a2593be0aad398546ca388ac"], ["created_at", "2021-07-11 16:06:59.064628"], ["updated_at", "2021-07-11 16:06:59.064628"]]
  TRANSACTION (0.2ms)  RELEASE SAVEPOINT active_record_1
  TRANSACTION (21.4ms)  commit transaction
=> 
[#<Glueby::Contract::Token:0x000055d24fc20480
...
irb(main):029:1* Glueby::Internal::RPC.perform_as("wallet-"+alice.id) do |client|
irb(main):030:1*       client.generate(1, authority_key)
irb(main):031:0> end
=> ["a0cf07a055a48c65d50233b6bc01f51d00c2bee5d63d2d78d971fb4b4df6a65a"]
irb(main):032:1* Glueby::Internal::RPC.perform_as("wallet-"+alice.id) do |client|
irb(main):033:1*       client.listunspent
irb(main):034:0> end
=> 
[{"txid"=>"23914e92f344cf617beabf87cb1df6902a895ad10e36da82ab3290eac6283712",
  "vout"=>0,
  "address"=>"mv2RncqVDmcs3tBv4dTouYiJMEmg8w6CmY",
  "token"=>"TPC",
  "amount"=>"50.00020000",
  "scriptPubKey"=>"2103e231d75a0057f76564b20306621b897fb930f7b68204e36e688327d867eb8686ac",
  "confirmations"=>1,
  "spendable"=>true,
  "solvable"=>true,
  "safe"=>true},
 {"txid"=>"4c7b75c78402142f560a8c11eebe5dd0f0ea26ceb77df9654bdedb84a1660960",
  "vout"=>1,
  "address"=>"mhLjKqXv2odQAWJbSeDqyu48BB1Hifw3BJ",
  "token"=>"TPC",
  "amount"=>"49.99980000",
  "scriptPubKey"=>"76a914140144093d8423f69caf96d9ecb8c66a60ba945488ac",
  "confirmations"=>1,
  "spendable"=>true,
  "solvable"=>true,
  "safe"=>true},
 {"txid"=>"746f2f232328334e1d0cb0bc5fe480c05807d967391d8142110aead4571f54c7",
  "vout"=>0,
  "address"=>"22VP6hHmzTMrvkkLUTMjB8JT1689UJmyoJM9FjawNVZ2q8T3nbgxKqVNRnYKRYsg53XYFTuJDFrpr5BQ",
  "token"=>"c1dc28d2651ee01745b1f112c0502db47434ebc5e2140b7ba2b324ca3ada3655b3",
  "amount"=>100,
  "scriptPubKey"=>"21c1dc28d2651ee01745b1f112c0502db47434ebc5e2140b7ba2b324ca3ada3655b3bc76a9148147ad309a550f723439a2593be0aad398546ca388ac",
  "confirmations"=>1,
  "spendable"=>true,
  "solvable"=>true,
  "safe"=>true}]
```

# Environment
#### OS
Mac OS Big Sure version 11.4

#### docker
Docker version 20.10.7, build f0df350
#### ruby
ruby 3.0.1p64 (2021-04-05 revision 0fb782ee38) [x86_64-darwin20]

#### tapyrusd docker image info

| REPOSITORY               | TAG              | IMAGE ID       | CREATED          | SIZE |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| tapyrus/tapyrusd         | edge             | a28e61cf8581   | 12 days ago      | 81.3MB |

### docker-compose.yml
```
version: "3"
services:
  tapyrusd:
    image: tapyrus/tapyrusd:edge
    environment:
      # genesis block:
      # $ docker run tapyrus/tapyrusd:latest tapyrus-genesis -dev -signblockpubkey=03af80b90d25145da28c583359beb47b21796b2fe1a23c1511e443e7a64dfdb27d -signblockprivatekey=cUJN5RVzYWFoeY8rUztd47jzXCu1p57Ay8V7pqCzsBD3PEXN7Dd4
      # How to create blocks.
      # $ docker-compose exec tapyrusd tapyrus-cli -conf=/etc/tapyrus/tapyrus.conf generatetoaddress 1 "pay-to-address" "c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3"
      GENESIS_BLOCK_WITH_SIG: 0100000000000000000000000000000000000000000000000000000000000000000000002b5331139c6bc8646bb4e5737c51378133f70b9712b75548cb3c05f9188670e7440d295e7300c5640730c4634402a3e66fb5d921f76b48d8972a484cc0361e66ef74f45e012103af80b90d25145da28c583359beb47b21796b2fe1a23c1511e443e7a64dfdb27d40e05f064662d6b9acf65ae416379d82e11a9b78cdeb3a316d1057cd2780e3727f70a61f901d10acbe349cd11e04aa6b4351e782c44670aefbe138e99a5ce75ace01010000000100000000000000000000000000000000000000000000000000000000000000000000000000ffffffff0100f2052a010000001976a91445d405b9ed450fec89044f9b7a99a4ef6fe2cd3f88ac00000000
    volumes:
#      - ./config/tapyrus.conf:/etc/tapyrus/tapyrus.conf
      - tapyrus:/var/lib/tapyrus
    ports:
      - "12381:12381"
  web:
    build: .
    tty: true
    volumes:
      - .:/myapp
      - ~/.ssh:/root/.ssh
      - gemfiles:/usr/local/bundle
    ports:
      - "3000:3000"
volumes:
  gemfiles:
  tapyrus:
```

### Dockerfile:
```
FROM ruby:3.0.1

RUN apt-get update -qq \
    && apt-get install -y nodejs libsodium-dev libpq-dev \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*
RUN mkdir /myapp

RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
RUN apt-get update && apt-get install -y yarn

WORKDIR /myapp

COPY Gemfile /myapp/Gemfile
COPY Gemfile.lock /myapp/Gemfile.lock

RUN gem install bundler
RUN gem install devise
RUN gem install nokogiri --platform=ruby
RUN bundle install

COPY . /myapp
CMD ["rails", "server", "-b", "0.0.0.0", "-p", "3000"]
```

### Gemfile
```
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '3.0.1'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails', branch: 'main'
gem 'rails', '~> 6.1.4'
# Use sqlite3 as the database for Active Record
gem 'sqlite3', '~> 1.4'
# Use Puma as the app server
gem 'puma', '~> 5.0'
# Use SCSS for stylesheets
gem 'sass-rails', '>= 6'
# Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
gem 'webpacker', '~> 5.0'
# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
gem 'turbolinks', '~> 5'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.7'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'
# Use Active Model has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Active Storage variant
# gem 'image_processing', '~> 1.2'

# Reduces boot times through caching; required in config/boot.rb
gem 'bootsnap', '>= 1.4.4', require: false

# For tapyrus gems
# Tapyrus low level tools library
gem 'tapyrus'
# Tapyrus High level library. make contract easily.
gem 'glueby'

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
end

group :development do
  # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
  gem 'web-console', '>= 4.1.0'
  # Display performance information such as SQL time and flame graphs for each request in your browser.
  # Can be configured to work on production as well see: https://github.com/MiniProfiler/rack-mini-profiler/blob/master/README.md
  gem 'rack-mini-profiler', '~> 2.0'
  gem 'listen', '~> 3.3'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
end

group :test do
  # Adds support for Capybara system testing and selenium driver
  gem 'capybara', '>= 3.26'
  gem 'selenium-webdriver'
  # Easy installation and use of web drivers to run system tests with browsers
  gem 'webdrivers'
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```

**Comment by azuchi** [comment]
Good catch. 

> I expected to Colored Coin scale is also 'TPC' unit that is same as UnColored Coin in listunspent.

However, the unit of colored coins is decided by the issuer, not by us. Therefore, conversion like TPC <-> tapyrus is not possible.
So we will fix the unit of colored coins in `getrawtransaction`.

**Comment by Naviabheeman** [comment]
Fixed in PR #147. Other RPC changes needed to support coloured coins are listed in #148
